src/viewport/viewport.service.ts

```diff
@@ -100,72 +52,50 @@ export class ViewportService {
 
 		this.sizeType$ = this.resize$.pipe(
 			startWith(this.getViewportSize()),
-			map(x => this.calculateViewportSize(x.width)),
+			distinctUntilChanged((a, b) => a.width === b.width),
+			map(x => this.getWidthSizeInfo(x.width)),
 			distinctUntilChanged(),
 			shareReplay(1),
 		);
 	}
 
 	/**
-	 * Calculates amount of items that fits into container's width.
-	 * @param containerWidth
-	 * @param itemWidth
-	 * @returns
+	 * Returns the current viewport size
 	 */
-	calculateItemsPerRow(containerWidth: number, itemWidth: number): number {
-		if (containerWidth === 0) {
-			return 0;
-		}
-
-		if (!containerWidth && !this.windowRef.hasNative) {
-			// todo: find a way to get container width for ssr
-			containerWidth = this.viewportServerSize.get().width;
-		}
-
-		return containerWidth / itemWidth;
-	}
-
```

Comment:
seemed redundant

Reply:
that was kinda an additional utility feature unrelated, so its best if its re-added

Reply:
i've removed it as it was only dividing the given width by an amount which can be done through `$sizeType` pipe on the app side.
will re-add though